pkg/kapp/resourcesmisc/apps_v1_deployment.go

```diff
@@ -75,6 +75,10 @@ func (s AppsV1Deployment) IsDoneApplying() DoneApplyState {
 			"Waiting for %d unavailable replicas", dep.Status.UnavailableReplicas)}
 	}
 
+	if dep.Spec.Replicas != nil && dep.Status.ReadyReplicas < *dep.Spec.Replicas {
```

Comment:
what happened to the nil check we were talking about?

Reply:
I think the pointer is not derefrenced if the first condition among these two return false?
`dep.Spec.Replicas != nil && dep.Status.ReadyReplicas < *dep.Spec.Replicas`
I think it is safe.

Reply:
@cppforlife  we were talking about the nil check of "dep.Spec.Replicas != nil" and as it is added with && operator it will safely exit if dep.Spec.Replicas is nil.